lens/util.go

```diff
@@ -0,0 +1,65 @@
+package lens
+
+import (
+	"context"
+	"errors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/chain/state"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/node/impl/full"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	"golang.org/x/xerrors"
+)
+
+var logger = logging.Logger("visor/lens/lotus")
+
+// OptimizedStateGetActorWithFallback is a helper to obtain an actor in the
+// state of the current tipset without recomputing the full tipset. It does
+// this by obtaining the child tipset (current height+1) and using the
+// pre-computed ParentState().
+//
+// TODO: Remove. See:  https://github.com/filecoin-project/sentinel-visor/issues/196
+func OptimizedStateGetActorWithFallback(ctx context.Context, api API, fallback full.StateModuleAPI, actor address.Address, tsk types.TipSetKey) (*types.Actor, error) {
+	act, err := efficientStateGetActor(ctx, api, actor, tsk)
+	if err != nil {
+		logger.Warnf("Optimized StateGetActorError: %s. Falling back to default StateGetActor().")
+		return fallback.StateGetActor(ctx, actor, tsk)
+	}
+	return act, nil
+}
+
+func efficientStateGetActor(ctx context.Context, api API, actor address.Address, tsk types.TipSetKey) (*types.Actor, error) {
+	ts, err := api.ChainGetTipSet(ctx, tsk)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to load tipset: %w", err)
+	}
+
+	child, err := api.ChainGetTipSetByHeight(ctx, ts.Height()+1, types.NewTipSetKey())
```

Comment:
I think this is missing handling for null rounds, I believe `child == ts` will be true for that case.

Reply:
looking

Reply:
I've been able to manually confirm it (that is `child == ts` when there is a null round after `ts) on my local deployment. I think the lotus chain store has a method for handling this case gracefully: https://github.com/filecoin-project/lotus/blob/master/chain/store/store.go#L1261

Reply:
I think this is resolved by the change you added right?

Reply:
This is resolved by using `GetTipsetByHeight` and passing `false` in, so yes I believe we can resolve this comment.